src/store/Network.ts

```diff
@@ -255,15 +255,18 @@ export default {
             const currentProfile: ProfileModel = rootGetters['profile/currentProfile'];
             const networkService = new NetworkService();
             const nodeService = new NodeService();
+            if (!newCandidate) {
+                newCandidate = networkConfig.nodes[Math.floor(Math.random() * networkConfig.nodes.length)].url;
+            }
             const networkModelResult = await networkService
                 .getNetworkModel(newCandidate, (currentProfile && currentProfile.generationHash) || undefined)
                 .toPromise();
             if (!networkModelResult) {
-                throw new Error('Connect error, active peer cannot be found');
+                await dispatch('CONNECT');
```

Comment:
there should be a throw I guess, e.g. user's own network problem. otherwise it is going to try indefinitely. 
there should be a final cut off and throw